### DISCUSSION TITLE
How does Frepple chose between suppliers and manufacturers?

### DISCUSSION THREAD

**Discussion by ThivaS357** [frePPLe/frepple#507] [Q&A]
Suppose I have an item that is both supplied by a supplier and produced by a manufacturer (i.e. an output of an operation)

1.How does Frepple prioritize between them?

2.Can I configure a percentage of an order that should be fulfilled from suppliers and a percentage from manufacturer?

**Comment by hichamlahlou** [comment]
Hello,

 

1.	Both item supplier and operation objects have a priority field. The solver will use the path with the highest priority (remember that priority 1 is higher than priority 10) as long as the demand is served on time. If highest priority path doesn’t allow to be on time, then the solver will revert to lower priority path if it allows to serve the demand on time. Note that https://demo.frepple.com has such an example with the chair that can be either purchased from the chair supplier or assembled in the factory. Assembly operation has priority 1 (we try to manufacture the chair as much as we can) and purchasing operation (item supplier record) has priority 2.
2.	You cannot do that between an item supplier and an operation. You can do it between multiple operations, refer to the split operation type in this example: https://frepple.org/docs/current/examples/operation/operation-type.php#

You can be creative and add a dummy operation to make it work.

I hope this helps.

 

Hicham

 

From: ThivaS357 ***@***.***> 
Sent: Monday, February 28, 2022 6:58 PM
To: frePPLe/frepple ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [frePPLe/frepple] How does Frepple chose between suppliers and manufacturers? (Discussion #507)

 

Suppose I have an item that is both supplied by a supplier and produced by a manufacturer (i.e. an output of an operation)

1.How does Frepple prioritize between them?

2.Can I configure a percentage of an order that should be fulfilled from suppliers and a percentage from manufacturer?

—
Reply to this email directly, view it on GitHub <https://github.com/frePPLe/frepple/discussions/507> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADLFAKKKQGVZB5YRG44PQ2LU5OZT7ANCNFSM5PRY2DPA> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> .

**Reply by djdkochasoft** [reply]
Hey there!. I found this interesting so I played around with the priorities:
What I noticed is that, it will adhere to the priority I have provided between item supplier and operation, ONLY when the location of both the operation and item supplier is the same.
When the locations are different, it will always try to prioritize manufacturing over supplying.

Is there a way to override this?
Thanks!!!

**Comment by hichamlahlou** [comment]
Hi,

 

The solver uses the priority when there are alternate replenishment possibilities at buffer level ***@***.***), not at item level in different locations.

So what you observe is valid and as expected.

If you have multiple locations, it means that your item is transported from location A (where you purchase it) to location B (where you manufacture it). The priority needs therefore to be set at item distribution level, not at item supplier level.

 

Hicham
